code/obj/machinery/alarm.dm

```diff
@@ -157,6 +158,18 @@
 	light_ov.plane = PLANE_SELFILLUM
 	src.UpdateOverlays(light_ov, "light")
 
+	if ((safe == ALARM_GOOD) && src.alertingAI)
+		for_by_tcl(aiPlayer, /mob/living/silicon/ai)
+			aiPlayer.cancelAlarm("Atmosphere", get_area(src), src)
+		src.alertingAI = FALSE
```

Comment:
```suggestion
	if (safe == ALARM_GOOD)
		if (src.alertingAI)
			for_by_tcl(aiPlayer, /mob/living/silicon/ai)
				aiPlayer.cancelAlarm("Atmosphere", get_area(src), src)
			src.alertingAI = FALSE
```
Otherwise if it's not currently alerting an AI it will send an alert even if the current state is `ALARM_GOOD`, probably causing your known issue?